Content.Server/Traits/Assorted/SelfAwareComponent.cs

```diff
@@ -0,0 +1,30 @@
+using Content.Shared.Damage.Prototypes;
+using Content.Shared.FixedPoint;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype.Set;
+
+namespace Content.Server.Traits.Assorted;
+
+/// <summary>
+/// This is used for the Self-Aware trait to enhance the information received from HealthExaminableSystem.
+/// </summary>
+[RegisterComponent]
+public sealed partial class SelfAwareComponent : Component
+{
+    // <summary>
+    // Damage types that an entity is able to precisely analyze like a health analyzer when they examine themselves.
+    // </summary>
+    [DataField("analyzableTypes", required: true, customTypeSerializer:typeof(PrototypeIdHashSetSerializer<DamageTypePrototype>))]
+    public HashSet<string> AnalyzableTypes = default!;
+
+    // <summary>
+    // Damage groups that an entity is able to detect the presence of when they examine themselves.
+    // </summary>
+    [DataField("detectableGroups", required: true, customTypeSerializer:typeof(PrototypeIdHashSetSerializer<DamageGroupPrototype>))]
+    public HashSet<string> DetectableGroups = default!;
+
+    // <summary>
+    // The thresholds for DetectableGroups.
+    // </summary>
+    public List<FixedPoint2> Thresholds = new()
+        { FixedPoint2.New(10), FixedPoint2.New(25), FixedPoint2.New(50), FixedPoint2.New(75) };
```

Comment:
Please don't hardcode the thresholds like this, especially since we have health as a variable. You should be instead checking these as percentages of the crit threshold, rather than a preset value. Both Health and Stamina Systems have extremely helpful functions for "Getting what is my percentage of the way to a specific threshold", that you can use. If someone's crit threshold is 135 instead of 100 for instance, these are now no longer correct to use.

Reply:
Got it. Should I make normal examinations take into account percentages of the crit threshold too while we're at it?

Reply:
If they're giving specific numbers it probably doesn't matter, but in the case of the poison category being seemingly a percentage of crit, then it should. Else the other categories can remain unchanged.

Reply:
I was referring to non-Self Aware examinations and examining other people

Reply:
Oh! Wait is that- seriously hardcoded around 100 crit by the base game!?
By all means be my guest and make that a percentage of crit threshold too lmao.

Reply:
Resolved with [604fd7d](https://github.com/Simple-Station/Einstein-Engines/pull/680/commits/604fd7d1a24bf1686bfb00e073b04fb1786e0173) and [e83bd60](https://github.com/Simple-Station/Einstein-Engines/pull/680/commits/e83bd60c3a11b2dbccdc1abc6d2ae59a844db58f)